sts_synindex_external.R

```diff
@@ -21,33 +21,24 @@ if (PARQUET_BUCKET==token$bucket && PARQUET_BUCKET_BASE_KEY==token$baseKey) {
   base_s3_uri <- paste0('s3://', PARQUET_BUCKET, '/', PARQUET_BUCKET_BASE_KEY)
 }
 
+base_s3_uri_external <- paste0('s3://', PARQUET_BUCKET_EXTERNAL, '/', PARQUET_BUCKET_BASE_KEY_EXTERNAL)
+base_s3_uri_archive <- paste0('s3://', PARQUET_BUCKET_EXTERNAL, '/', PARQUET_BUCKET_BASE_KEY_ARCHIVE)
+
 # configure the environment with AWS token
 Sys.setenv('AWS_ACCESS_KEY_ID'=token$accessKeyId,
            'AWS_SECRET_ACCESS_KEY'=token$secretAccessKey,
            'AWS_SESSION_TOKEN'=token$sessionToken)
 
 #### Sync bucket to local dir####
+unlink(AWS_PARQUET_DOWNLOAD_LOCATION, recursive = T, force = T)
 sync_cmd <- glue::glue('aws s3 sync {base_s3_uri} {AWS_PARQUET_DOWNLOAD_LOCATION} --exclude "*owner.txt*" --exclude "*archive*"')
 system(sync_cmd)
 
-#### Drop columns with potentially identifying info ####
-drop_cols_datasets <- function(dataset, columns=c(), output=PARQUET_FILTERED_LOCATION) {
-  if (dataset %in% list.dirs(AWS_PARQUET_DOWNLOAD_LOCATION, full.names = F)) {
-    input_path <- paste0(AWS_PARQUET_DOWNLOAD_LOCATION, '/', dataset)
-    final_path <- paste0(output, '/', dataset, '/')
-    
-    arrow::open_dataset(sources = input_path) %>% 
-      dplyr::select(!columns) %>% 
-      arrow::write_dataset(path = final_path, max_rows_per_file = 900000)
-  }
-}
 
-lapply(seq_along(datasets_to_filter), function(i) {
-  cat("Dropping ", cols_to_drop[[i]], " from ", datasets_to_filter[i], "\n")
-  drop_cols_datasets(dataset = datasets_to_filter[i], columns = cols_to_drop[[i]])
-})
+# Filter parquet datasets -------------------------------------------------
+source('~/recover-parquet-external/filtering.R')
```

Comment:
Here you are actually executing the filters.  You may want to think about having one "workflow" script that sources all the functions and calls the functions.  This way you're not having different R scripts actually calling code.  So example:

```
source('filter_functions')
run_my_filter_here()
```